app/src/main/java/org/wikipedia/travel/landmarkpicker/LandmarkFragment.java

```diff
@@ -136,22 +138,42 @@ private void onLoadPage(@NonNull PageTitle title, HistoryEntry entry) {
         }
     }
 
-    private String onRequestOpenDestinationName(){
+    private Trip.Destination onRequestOpenDestination() {
         Callback callback = getCallback();
         if (callback != null) {
-            callback.onRequestOpenDestinationName();
+            return callback.onRequestOpenDestination();
         }
-            return "";
+        return null;
     }
 
-    //Saves list of selectedLandmarks
-    private void onSave(List<LandmarkCard> saveList){
-        Callback callback = getCallback();
-        if (callback != null) {
-            callback.onSave(saveList);
+    private void loadSelectedCards() {
+        if (this.destination != null) {
+            if (getCallback() != null) {
```

Comment:
`getCallback()` should ideally only be invoked once, so it should be initially stored in a variable before the conditional, and every instance of `getCallback()` in this function should use that variable